inc/database/class-tmsc-media.php

```diff
@@ -0,0 +1,301 @@
+
+			// Update status.
+			$this->after_save();
+
+			return true;
+		}
+		*/
+		return false;
+	}
+
+	/**
+	 * Save post
+	 * @return WP_Object Object
+	 */
+	public function save_post() {
+
+		// $filename should be the path to a file in the upload directory.
+		$filename = $;
+
+		// The ID of the post this attachment is for.
+		$parent_post_id = 37;
+
+		// Check the type of file. We'll use this as the 'post_mime_type'.
+		$filetype = wp_check_filetype( basename( $filename ), null );
+
+		// Get the path to the upload directory.
+		$wp_upload_dir = wp_upload_dir();
+
+		// Prepare an array of post data for the attachment.
+		$attachment = array(
+			'guid'           => $wp_upload_dir['url'] . '/' . basename( $filename ),
+			'post_mime_type' => $filetype['type'],
+			'post_title'     => preg_replace( '/\.[^.]+$/', '', basename( $filename ) ),
+			'post_content'   => '',
+			'post_status'    => 'inherit',
+		);
+
+		// Insert the attachment.
+		$attach_id = wp_insert_attachment( $attachment, $filename, $parent_post_id );
+
+
+		$date = date( 'Y-m-d H:i:s', $this->get_pubdate() );
+		$post = array(
+			'ID' => empty( $this->object->ID ) ? 0 : $this->object->ID,
+			'post_title' => $this->get_title(),
+			'post_status' => 'migrating',
+			'post_author' => $this->get_post_author(),
+			'post_date' => $date,
+			'post_date_gmt' => get_gmt_from_date( $date ),
+			'post_type' => $this->get_post_type(),
+			'post_content' => $this->get_body(),
+			'post_excerpt' => $this->get_excerpt(),
+			'post_name' => $this->get_post_name(),
+			'comment_status' => 'closed',
+		);
+
+		$post_id = wp_insert_post( $post );
+		if ( ! empty( $post_id ) ) {
+			return get_post( $post_id );
+		}
+		return false;
+	}
+
+	/**
+	 * Save post meta data
+	 * @return void
+	 */
+	public function save_meta_data() {
+		if ( ! empty( $this->object->ID ) ) {
+			// Get our meta data mapping and iterate through it.
+			foreach ( $this->get_meta_keys() as $key => $db_field ) {
+				$this->update_meta( $key, $this->raw->$db_field );
+			}
+		}
+		return;
+	}
+
+	/**
+	 * Map our raw data keys to our meta keys
+	 * @return array. An array of post meta keys and corresponding db fields in our raw data.
+	 */
+	public function get_meta_keys() {
+		return apply_filters( 'tmsc_media_meta_keys', array() );
+	}
+
+	/**
+	 * Save media attachments.
+	 */
+	public function save_media_attachments() {
+		if ( ! empty( $this->object->ID ) && ! empty( $this->raw->ObjectID ) ) {
+			$attachments = $this->processor->get_object_attachments( $this->raw->ObjectID );
+			foreach ( $attachments as $attachment_raw_data ) {
+				$this->add_attachment( $attachment_raw_data, $this->object->ID );
+			}
+		}
+	}
+
+
+	/**
+	 * Create or update an attachment
+	 * @return object. WP_Attachment
+	 */
+	public function add_attachment( $data, $object_id ) {
+
+		// $filename should be the path to a file in the upload directory.
+		$filename = '/path/to/uploads/2013/03/filename.jpg';
```

Comment:
This hard-coded value seems to be for debugging or something else, please investigate.